include/realizations/catchment/Bmi_Formulation.hpp

```diff
@@ -26,6 +26,10 @@
 #define BMI_REALIZATION_CFG_PARAM_OPT__LIB_FILE "library_file"
 #define BMI_REALIZATION_CFG_PARAM_OPT__PYTHON_TYPE_NAME "python_type"
 #define BMI_REALIZATION_CFG_PARAM_OPT__REGISTRATION_FUNC "registration_function"
+#define BMI_REALIZATION_CFG_PARAM_OPT__CPP_CREATE_FUNC "create_function"
```

Comment:
Yes, very similar...I am inclined to leave this one named `create` because that implies that you need to do `destroy` (/`delete`), where there's no such antonym on the C side...but it's subjective.

Reply:
Doubling down on the naming differences. For C (and Fortran?) the framework allocates the memory that is populated by the "register" function, and is responsible for freeing it. For C++, the create function in the module is responsible for allocating the object and the destroy function is responsible for deleting it. Just the difference on the create side is I think enough to justify not calling it "register".

Reply:
🤷‍♂️

Reply:
I'll interpret that as consent... Yeah, spent some more time thinking about it, and we can't really do something comparable to C/Fortran... we can't create an "empty" `bmi::Bmi` class and hand it to a function to be "populated", the concrete class has to be created/constructed on the module side. The `register` paradigm exists to handle the language construct differences--it just _happens_ to provide a good access point for accessing code in the DLL. In the C++ case the interaction across the DLL boundary has to be different so I think the difference in nomenclature is needed.